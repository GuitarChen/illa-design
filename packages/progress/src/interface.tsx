import { HTMLAttributes, ReactNode } from "react"

export type ProgressColorScheme =
  | string
  | "white"
  | "blackAlpha"
  | "gray"
  | "grayBlue"
  | "red"
  | "orange"
  | "yellow"
  | "green"
  | "blue"
  | "cyan"
  | "purple"
  | "techPink"
  | "techPurple"

export type ProgressType = "line" | "circle" | "miniCircle" | "miniRing"

export type ProgressStatus = "normal" | "success" | "error"

export type ProgressSize = "small" | "medium" | "large"

export interface ProgressProps extends HTMLAttributes<HTMLDivElement> {
  type?: ProgressType
  steps?: number
  status?: ProgressStatus
  color?: ProgressColorScheme
  trailColor?: ProgressColorScheme
  showText?: boolean
  formatText?: (percent: number) => ReactNode
  percent?: number
  width?: string
  strokeWidth?: string
}
