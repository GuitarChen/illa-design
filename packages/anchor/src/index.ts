export * from "./anchor"
export * from "./interface"
