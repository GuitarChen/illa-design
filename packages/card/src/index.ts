export * from "./card"
export * from "./interface"
export * from "./card-grid"
export * from "./meta"
