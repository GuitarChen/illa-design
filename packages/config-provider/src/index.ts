export * from "./config-provider"
export * from "./interface"
export * from "./config-provider-context"
export * from "./config"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
