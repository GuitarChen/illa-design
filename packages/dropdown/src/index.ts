export * from "./dropdown"
export * from "./interface"
