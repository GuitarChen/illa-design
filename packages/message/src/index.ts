export * from "./style"
export * from "./message"
export * from "./interface"
