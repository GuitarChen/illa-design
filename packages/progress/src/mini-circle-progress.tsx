import { forwardRef, ReactNode } from "react"
import { ProgressProps } from "./interface"
import { SuccessIcon, WarningCircleIcon } from "@illa-design/icon"
import { Trigger } from "@illa-design/trigger"
import { applyContainer } from "./common-style"
import {
  applyCircleProgressContainer,
  applyCircleStatus,
  applyCircleSvgContainer,
} from "./mini-circle-progress-style"
import { globalColor, illaPrefix } from "@illa-design/theme"

export const MiniCircleProgress = forwardRef<HTMLDivElement, ProgressProps>(
  (props, ref) => {
    const {
      type = "line",
      status = "normal",
      color = "blue",
      width = "16px",
      strokeWidth = "4px",
      trailColor = "gray",
      showText = true,
      formatText = (percent: number) => {
        return `${percent}%`
      },
      percent = 0,
      steps,
      ...otherProps
    } = props

    const radius = `calc(${width} / 2)`
    const center = `calc(${width} / 2)`

    let finalColor: string

    switch (status) {
      case "normal":
        finalColor = color
        break
      case "success":
        finalColor = "green"
        break
      case "error":
        finalColor = "red"
        break
    }

    let statusComponent: ReactNode
    switch (status) {
      case "success":
        statusComponent = (
          <SuccessIcon color={globalColor(`--${illaPrefix}-green-03`)} />
        )
        break
      case "error":
        statusComponent = (
          <WarningCircleIcon color={globalColor(`--${illaPrefix}-red-03`)} />
        )
        break
    }

    return (
      <div css={applyContainer(width)} ref={ref} {...otherProps}>
        <Trigger disabled={!showText} content={formatText(percent)}>
          {(status == "normal" || status == "error") && (
            <svg css={applyCircleSvgContainer(width, trailColor)}>
              <circle
                css={applyCircleProgressContainer(finalColor, width, percent)}
                fill="none"
                cx={center}
                cy={center}
                r={radius}
                strokeWidth={width}
              />
            </svg>
          )}
          {status == "success" && (
            <span css={applyCircleStatus}>{statusComponent}</span>
          )}
        </Trigger>
      </div>
    )
  },
)

MiniCircleProgress.displayName = "MiniCircleProgress"
