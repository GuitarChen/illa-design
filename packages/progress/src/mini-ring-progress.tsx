import { forwardRef } from "react"
import { ProgressProps } from "./interface"
import { Trigger } from "@illa-design/trigger"
import {
  applyContainer,
  applyProgressContainer,
  applyProgressContainerBg,
  applySvgContainer,
} from "./common-style"

export const MiniRingProgress = forwardRef<HTMLDivElement, ProgressProps>(
  (props, ref) => {
    const {
      type = "line",
      steps,
      status = "normal",
      color = "blue",
      width = "16px",
      strokeWidth = "4px",
      trailColor = "gray",
      showText = true,
      formatText = (percent: number) => {
        return `${percent}%`
      },
      percent = 0,
      ...otherProps
    } = props

    const radius = `calc((${width} - ${strokeWidth}) / 2)`
    const center = `calc(${width} / 2)`

    let finalColor: string

    switch (status) {
      case "normal":
        finalColor = color
        break
      case "success":
        finalColor = "green"
        break
      case "error":
        finalColor = "red"
        break
    }

    return (
      <div css={applyContainer(width)} ref={ref} {...otherProps}>
        <Trigger disabled={!showText} content={formatText(percent)}>
          <svg css={applySvgContainer(width)}>
            <circle
              css={applyProgressContainerBg(trailColor, width, strokeWidth)}
              fill="none"
              cx={center}
              cy={center}
              r={radius}
              strokeWidth={strokeWidth}
            />
            <circle
              css={applyProgressContainer(
                finalColor,
                width,
                strokeWidth,
                percent,
              )}
              fill="none"
              cx={center}
              cy={center}
              r={radius}
              strokeWidth={strokeWidth}
            />
          </svg>
        </Trigger>
      </div>
    )
  },
)

MiniRingProgress.displayName = "MiniRingProgress"
