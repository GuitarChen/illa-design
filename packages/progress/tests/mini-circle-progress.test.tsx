import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import { Progress } from "../src"

test("Progress renders correctly", () => {
  render(
    <div>
      <Progress data-testid="test-progress" type="miniCircle" />
    </div>,
  )
  expect(screen.getByTestId("test-progress")).toBeInTheDocument()
})

test("Progress renders with colors", () => {
  render(
    <div>
      <Progress
        data-testid="test-progress-red"
        type="miniCircle"
        color="red"
        trailColor="red"
      />
      ,
      <Progress
        data-testid="test-progress-gray"
        type="miniCircle"
        color="gray"
        trailColor="gray"
      />
      ,
      <Progress
        data-testid="test-progress-custom"
        type="miniCircle"
        color="#123123"
        trailColor="#321321"
      />
      ,
    </div>,
  )
  expect(screen.getByTestId("test-progress-red")).toMatchSnapshot()
  expect(screen.getByTestId("test-progress-gray")).toMatchSnapshot()
  expect(screen.getByTestId("test-progress-custom")).toMatchSnapshot()
})

test("Progress renders with success status", () => {
  render(<Progress status="success" type="miniCircle" />)
  expect(screen.getByTitle("SuccessIcon")).toBeInTheDocument()
})

test("Progress renders with error status", () => {
  render(
    <Progress data-testid="test-progress" status="error" type="miniCircle" />,
  )
  expect(screen.getByTestId("test-progress")).toMatchSnapshot()
})

test("Progress renders without text", () => {
  render(<Progress percent={50} showText={false} type="miniCircle" />)
  expect(screen.queryByText("50")).not.toBeTruthy()
})

test("Progress renders with custom width", () => {
  render(
    <Progress
      data-testid="test-progress"
      percent={50}
      width="100px"
      type="miniCircle"
      strokeWidth="10px"
    />,
  )
  expect(screen.getByTestId("test-progress")).toMatchSnapshot()
})

test("Progress renders with steps", () => {
  render(<Progress data-testid="test-progress" steps={3} type="miniCircle" />)
  expect(screen.getByTestId("test-progress")).toMatchSnapshot()
})
