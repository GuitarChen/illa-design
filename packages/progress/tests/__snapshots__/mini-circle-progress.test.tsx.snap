// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders with colors 1`] = `
.emotion-0 {
  position: relative;
  width: 16px;
  height: 16px;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1 {
  display: inline-flex;
}

.emotion-2 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #fcc;
}

.emotion-3 {
  -webkit-transition: all 200ms;
  transition: all 200ms;
  stroke: #ff4747;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  transform-origin: center;
  stroke-dasharray: calc(3.141592653589793 * 2 * 16px / 2);
  stroke-dashoffset: calc(calc(3.141592653589793 * 2 * 16px / 2) - (0 / 100 * calc(3.141592653589793 * 2 * 16px / 2)));
}

<div
  class="emotion-0"
  data-testid="test-progress-red"
>
  <span
    class="emotion-1"
  >
    <svg
      class="emotion-2"
    >
      <circle
        class="emotion-3"
        cx="calc(16px / 2)"
        cy="calc(16px / 2)"
        fill="none"
        r="calc(16px / 2)"
        stroke-width="16px"
      />
    </svg>
  </span>
</div>
`;

exports[`Progress renders with colors 2`] = `
.emotion-0 {
  position: relative;
  width: 16px;
  height: 16px;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1 {
  display: inline-flex;
}

.emotion-2 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #ebebeb;
}

.emotion-3 {
  -webkit-transition: all 200ms;
  transition: all 200ms;
  stroke: #1f1f1f;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  transform-origin: center;
  stroke-dasharray: calc(3.141592653589793 * 2 * 16px / 2);
  stroke-dashoffset: calc(calc(3.141592653589793 * 2 * 16px / 2) - (0 / 100 * calc(3.141592653589793 * 2 * 16px / 2)));
}

<div
  class="emotion-0"
  data-testid="test-progress-gray"
>
  <span
    class="emotion-1"
  >
    <svg
      class="emotion-2"
    >
      <circle
        class="emotion-3"
        cx="calc(16px / 2)"
        cy="calc(16px / 2)"
        fill="none"
        r="calc(16px / 2)"
        stroke-width="16px"
      />
    </svg>
  </span>
</div>
`;

exports[`Progress renders with colors 3`] = `
.emotion-0 {
  position: relative;
  width: 16px;
  height: 16px;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1 {
  display: inline-flex;
}

.emotion-2 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #321321;
}

.emotion-3 {
  -webkit-transition: all 200ms;
  transition: all 200ms;
  stroke: #123123;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  transform-origin: center;
  stroke-dasharray: calc(3.141592653589793 * 2 * 16px / 2);
  stroke-dashoffset: calc(calc(3.141592653589793 * 2 * 16px / 2) - (0 / 100 * calc(3.141592653589793 * 2 * 16px / 2)));
}

<div
  class="emotion-0"
  data-testid="test-progress-custom"
>
  <span
    class="emotion-1"
  >
    <svg
      class="emotion-2"
    >
      <circle
        class="emotion-3"
        cx="calc(16px / 2)"
        cy="calc(16px / 2)"
        fill="none"
        r="calc(16px / 2)"
        stroke-width="16px"
      />
    </svg>
  </span>
</div>
`;

exports[`Progress renders with custom width 1`] = `
.emotion-0 {
  position: relative;
  width: 100px;
  height: 100px;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1 {
  display: inline-flex;
}

.emotion-2 {
  width: 100px;
  height: 100px;
  border-radius: 50%;
  background-color: #ebebeb;
}

.emotion-3 {
  -webkit-transition: all 200ms;
  transition: all 200ms;
  stroke: #1e6fff;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  transform-origin: center;
  stroke-dasharray: calc(3.141592653589793 * 2 * 100px / 2);
  stroke-dashoffset: calc(calc(3.141592653589793 * 2 * 100px / 2) - (50 / 100 * calc(3.141592653589793 * 2 * 100px / 2)));
}

<div
  class="emotion-0"
  data-testid="test-progress"
>
  <span
    class="emotion-1"
  >
    <svg
      class="emotion-2"
    >
      <circle
        class="emotion-3"
        cx="calc(100px / 2)"
        cy="calc(100px / 2)"
        fill="none"
        r="calc(100px / 2)"
        stroke-width="100px"
      />
    </svg>
  </span>
</div>
`;

exports[`Progress renders with error status 1`] = `
.emotion-0 {
  position: relative;
  width: 16px;
  height: 16px;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1 {
  display: inline-flex;
}

.emotion-2 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #ebebeb;
}

.emotion-3 {
  -webkit-transition: all 200ms;
  transition: all 200ms;
  stroke: #ff4747;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  transform-origin: center;
  stroke-dasharray: calc(3.141592653589793 * 2 * 16px / 2);
  stroke-dashoffset: calc(calc(3.141592653589793 * 2 * 16px / 2) - (0 / 100 * calc(3.141592653589793 * 2 * 16px / 2)));
}

<div
  class="emotion-0"
  data-testid="test-progress"
>
  <span
    class="emotion-1"
  >
    <svg
      class="emotion-2"
    >
      <circle
        class="emotion-3"
        cx="calc(16px / 2)"
        cy="calc(16px / 2)"
        fill="none"
        r="calc(16px / 2)"
        stroke-width="16px"
      />
    </svg>
  </span>
</div>
`;

exports[`Progress renders with steps 1`] = `
.emotion-0 {
  position: relative;
  width: 16px;
  height: 16px;
  vertical-align: middle;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-1 {
  display: inline-flex;
}

.emotion-2 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: #ebebeb;
}

.emotion-3 {
  -webkit-transition: all 200ms;
  transition: all 200ms;
  stroke: #1e6fff;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  transform-origin: center;
  stroke-dasharray: calc(3.141592653589793 * 2 * 16px / 2);
  stroke-dashoffset: calc(calc(3.141592653589793 * 2 * 16px / 2) - (0 / 100 * calc(3.141592653589793 * 2 * 16px / 2)));
}

<div
  class="emotion-0"
  data-testid="test-progress"
>
  <span
    class="emotion-1"
  >
    <svg
      class="emotion-2"
    >
      <circle
        class="emotion-3"
        cx="calc(16px / 2)"
        cy="calc(16px / 2)"
        fill="none"
        r="calc(16px / 2)"
        stroke-width="16px"
      />
    </svg>
  </span>
</div>
`;
